Sourcery/Parsing/FileParser.swift

```diff
@@ -307,6 +312,73 @@ extension FileParser {
         return true
     }
 
+    fileprivate func extractGenericsDeclaration(source: String) -> String {
+        var declaration = source
+        if let whereStart = declaration.index(of: "where") {
+            declaration = String(declaration[declaration.startIndex ..< whereStart])
+        }
+
+        if let start = source.index(of: "<") {
+            var end = start
+            var openingAngles = 1
+            while openingAngles > 0 && end != source.endIndex {
+                end = source.index(after: end)
+                if source[end] == ">" {
```

Comment:
is it really possible to have nested generic blocks?

Reply:
Yes, if they appear on the constraints. Something like: `class Foo<A, T : Bar<A>>` is syntactically correct.